WordPress/Classes/ViewRelated/Activity/ActivityListViewModel.swift

```diff
@@ -117,6 +135,38 @@ class ActivityListViewModel: Observable {
         // showing plugin updates/CTA's and other things like this.
     }
 
+    func dateRangeDescription() -> String? {
+        guard after != nil || before != nil else {
+            return nil
+        }
+
+        let format = isYearDifferent(between: after, and: before) ? "MMM d, yyyy" : "MMM d"
+        dateFormatter.setLocalizedDateFormatFromTemplate(format)
+
+        var formattedDateRanges: [String] = []
+
+        if let after = after {
+            formattedDateRanges.append(dateFormatter.string(from: after))
+        }
+
+        if let before = before {
+            formattedDateRanges.append(dateFormatter.string(from: before))
+        }
+
+        return formattedDateRanges.joined(separator: " - ")
+    }
+
+    private func isYearDifferent(between firstDate: Date?, and secondDate: Date?) -> Bool {
+        guard let firstDate = firstDate, let secondDate = secondDate else {
+            return false
+        }
+
+        let firstDateComponents = Calendar.current.dateComponents([.year], from: firstDate)
+        let secondDateComponents = Calendar.current.dateComponents([.year], from: secondDate)
+
+        return firstDateComponents.year != secondDateComponents.year
+    }
```

Comment:
Unfortunately this logic doesn't work if I select 2 dates from a different year.  Here's how I envisioned the logic working. 

Also a naming change suggestion. 
```suggestion
    private func shouldDisplayFullYear(with firstDate: Date?, and secondDate: Date?) -> Bool {
        guard let firstDate = firstDate, let secondDate = secondDate else {
            return false
        }

        let currentYear = Calendar.current.dateComponents([.year], from: Date()).year
        let firstYear = Calendar.current.dateComponents([.year], from: firstDate).year
        let secondYear = Calendar.current.dateComponents([.year], from: secondDate).year

        return firstYear != currentYear || secondYear != currentYear
    }
```

Reply:
Thanks for that! Done!